bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/client/MagentoGraphqlClientImpl.java

```diff
@@ -270,6 +280,14 @@ private static GraphqlClient adaptToGraphqlClient(Resource resource) {
         return graphqlClient;
     }
 
+    private static GraphqlResponse<Query, Error> newErrorResponse(Throwable throwable) {
+        GraphqlResponse<Query, Error> response = new GraphqlResponse<>();
+        Error error = new Error();
+        error.setMessage(throwable.getMessage());
```

Comment:
One thing we may consider here is to use the category property of the error object to set a special predefined category when the error represents an exception instead of a GraphQL error which is its original purpose. That can be used then for special error handling scenarios if needed also by customers and can be part of the Javadoc of MagentoGrapqlClient.

Reply:
Thats a good idea. Will do that.

Reply:
Done in https://github.com/adobe/aem-core-cif-components/pull/641/commits/36299db78dcdbc1abd07485134d1f1c748477722

Reply:
Thanks you, LGTM.